pkg/odo/genericclioptions/runnable.go

```diff
@@ -44,30 +44,36 @@ func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
 	startTime := time.Now()
 	cfg, _ := preference.NewClient()
 	disableTelemetry, _ := strconv.ParseBool(os.Getenv(segment.DisableTelemetryEnv))
+	debugTelemetry := segment.GetDebugTelemetry()
 
 	// Prompt the user to consent for telemetry if a value is not set already
 	// Skip prompting if the preference command is called
 	// This prompt has been placed here so that it does not prompt the user when they call --help
-
-	if !cfg.IsSet(preference.ConsentTelemetrySetting) && cmd.Parent().Name() != "preference" {
-		if !segment.RunningInTerminal() {
-			klog.V(4).Infof("Skipping telemetry question because there is no terminal (tty)\n")
-		} else if disableTelemetry {
-			klog.V(4).Infof("Skipping telemetry question due to %s=%t\n", segment.DisableTelemetryEnv, disableTelemetry)
-		} else {
-			var consentTelemetry bool
-			prompt := &survey.Confirm{Message: "Help odo improve by allowing it to collect usage data. Read about our privacy statement: https://developers.redhat.com/article/tool-data-collection. You can change your preference later by changing the ConsentTelemetry preference.", Default: true}
-			err = survey.AskOne(prompt, &consentTelemetry, nil)
-			ui.HandleError(err)
-			if err == nil {
-				if err1 := cfg.SetConfiguration(preference.ConsentTelemetrySetting, strconv.FormatBool(consentTelemetry)); err1 != nil {
-					klog.V(4).Info(err1.Error())
+	if len(debugTelemetry) == 0 {
```

Comment:
This check seems unnecessary. Why do you want to check it here?

Reply:
I think the way this should work is, if `ODO_DEBUG_TELEMETRY_FILE` is set, we write to a file, else we let segment do it's thing.
For tests, we export 2 variables, `ODO_DISABLE_TELEMETRY=false`, and `ODO_DEBUG_TELEMETRY_FILE=/some/file`.

Reply:
ok lemme see if i can bring the logic closer to that

Reply:
> I think the way this should work is, if `ODO_DEBUG_TELEMETRY_FILE` is set, we write to a file, else we let segment do it's thing. For tests, we export 2 variables, `ODO_DISABLE_TELEMETRY=false`, and `ODO_DEBUG_TELEMETRY_FILE=/some/file`.

By this, I mean, we leave the segment enable/disable logic as it is. The only thing that should be added here is logic with `ODO_DEBUG_TELEMETRY_FILE`. If it is set and if segment is enabled, we write to it.